File: samples/petstore/php/tests/StoreApiTest.php
<?php

require_once 'BaseApiTest.php';

class StoreApiTest extends BaseApiTest {

  public function testGetOrderById() {
    $res = $this->storeApi->getOrderById(1);
    $this->assertEquals(1, $res->petId);
    $this->assertEquals('DateTime', get_class($res->shipDate));
  }

  public function testDeleteOrder() {
    $res = $this->storeApi->deleteOrder(3);
    $res = $this->storeApi->deleteOrder("foo");

    // We just want to make sure there are no errors in this test.
    // To verify you are getting back a 200, you might want to add
    // something like this to Swagger.php callAPI():
    // print "Response for call to $resourcePath : ";
    // print_r($data);
  }

  public function testPlaceOrder() {
    $order = new Order();

    $order->petId = 1;
    $order->status = "ordered";
    $order->quantity = 10;
    // $order->shipDate = "1/1/2013";

    $res = $this->storeApi->placeOrder($order);

    // We just want to make sure there are no errors in this test.
    // To verify you are getting back a 200, you might want to add
    // something like this to Swagger.php callAPI():
    // print "Response for call to $resourcePath : ";
    // print_r($data);
  }

}
?>

File: samples/petstore/python/tests/StoreApiTest.py
#!/usr/bin/env python

import sys
import unittest
import datetime

from BaseApiTest import BaseApiTest

sys.path = ['./'] + sys.path
from petstore import *
from petstore.models import *


class StoreApiTest(BaseApiTest):

    def testGetOrderById(self):
        res = self.storeApi.getOrderById(1)
        assert res, 'null getOrderById result'
        assert long(1) == res.id, 'order id should be long(1)'

    def testDeleteOrder(self):
        self.storeApi.deleteOrder(3)
        self.storeApi.deleteOrder("foo")

    def testPlaceOrder(self):
        order = Order.Order()
        order.petId = 1
        order.status = 'ordered'
        order.quantity = 10
        order.shipDate = datetime.datetime.strptime("2011-01-09T13:55:07.123",
                                                    "%Y-%m-%dT%H:%M:%S.%f")
        self.storeApi.placeOrder(order)


if __name__ == "__main__":
    unittest.main()
